src/xunit.v3.core.tests/Acceptance/FixtureAcceptanceTests.cs

```diff
@@ -544,6 +544,35 @@ public ClassWithCountedFixture(CountedFixture fixture)
 			[Fact]
 			public void TheTest() { }
 		}
+
+		[Fact]
+		public async ValueTask CollectionFixtureOnGenericTestClassAcceptsArgument()
+		{
+			var messages = await RunAsync<_TestPassed>(typeof(GenericTests));
+
+			Assert.Equal(2, messages.Count);
+		}
+
+		[CollectionDefinition("generic collection")]
+		public class GenericFixtureCollection<T> : ICollectionFixture<GenericFixture<T>> { }
+
+		[Collection("generic collection")]
+		abstract class GenericTestBase<T>
+		{
+			protected GenericTestBase(GenericFixture<T> fixture) => Fixture = fixture;
+			protected readonly GenericFixture<T> Fixture;
+		}
+
+		public class GenericArgument { }
+
+		class GenericTests : GenericTestBase<GenericArgument>
+		{
+#pragma warning disable xUnit1041 // Fixture arguments to test classes must have fixture sources
```

Comment:
We'll want to make sure xUnit1041 gets updated to understand the new functionality.